### PR TITLE
Pl

### PR DESCRIPTION
**Pull request by justinchiu**
migrate to lightning and clean up loss computation

run [here](https://wandb.ai/chiu-justin-t/ssm-cifar-tokenized/jobs/QXJ0aWZhY3RDb2xsZWN0aW9u), truncated after 10 training steps.

the validation bpd logging is a little weird. if you see the bug, please help (resolved. lightning runs a few validation steps at the start of training as a sanity check)

I'm also not sure about checkpointing (resolved) and the learning rate monitor, main.py:50-51.

### CODE REVIEW

File: models/mamba.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import torch
+import torch.nn.functional as F
+import lightning as L
+import wandb
+import einops
+import torchvision
+
+from mamba_ssm.models.mixer_seq_simple import MambaLMHeadModel
+from mamba_ssm.utils.generation import decode
+from constants import LOG2, VOCAB_SIZE
```
Comment (Skylion007) [inline models/mamba.py]: Let's not make VOCAB_SIZE a global constant? We want the library to be more general and potentially work on grayscale datasets.

Comment (justinchiu) [reply models/mamba.py]: there are no density estimation results for grayscale though right? does imagenet also 256 values for each channel
